Recruit-Training/Embedded-Recruit-Training/ReadMe.adoc

```diff
@@ -0,0 +1,140 @@
+Embedded Recruit Training 
+------------------------
+
+In this recruit training you will be receiving two “control” CAN messages, one for the blue LED and the green LED. Depending on which bits are ‘set’ you will toggle either the blue or green LEDS. You will also send back a message with current state of the LEDs at a frequency of 1hz for the blue LED and 10Hz for the green. 
+
+We will begin by discussing the directory structure for the recruit training project.
+
+*  Core: where all of the user/main code will go
+** inc: where all of the header files go 
+** src: where all of the source files go 
+* Drivers: where all the HAL driver functions go
+* Middlewares: where all of the os functions (FREERTOS and CMSIS) are
+
+If there are any functions that you think you need, but aren't given, refer to Documentation/HALFunctionReferences.adoc and Documentation/RTOSReferences.adoc
+
+
+Start by going to Core/Src/main.c and search for “TODO” this will show you where you need to make changes
+
+. Within “USER CODE BEGIN Includes” and “USER CODE END Includes”
+** Include all task header files, these will be found under Core/Inc and have “Task” in the filename
+. Within “USER CODE BEGIN PV” and “USER CODE END PV”
+** Add the StdId for the green led, this value will be 0XBBB, look to blue StdId for an example
+** Add CAN Tx Header definition. The type is `CAN_TxHeaderTypeDef`
+** Define threads for the blue and green toggle LED tasks. The type will be `osThreadId_t`
+** Define toggle flags for blue and green LEDs. The type will be `uint8_t`
+. Within “USER CODE BEGIN 2” and “USER CODE END 2”
+** Call the function `MX_CAN2_UserInit`, it takes no arguments
+. Within “USER CODE BEGIN RTOS MUTEX” and “USER CODE END RTOS MUTEX”
+** Create mutexes for CAN transmit
+*** To create a mutex:
+**** Define a mutex handle of `osMutexId_t`
+**** Define attributes of const `osMutexAttr_t`
+***** This is a struct and will be initialized with:
+* String name 
+* Zero
+* Null
+* Zero 
+***** Ex:
+ `const osMutexAttr_t MutexAttr =
+   {
+       "MutexAttr", // name of mutex
+       0, // TODO perhaps osMutexRecursive
+       NULL,
+       0
+   };`
```

Comment:
Formatting broke for me, try using 
\`\`\`
<code>
\`\`\`

instead of the backticks